test/functionalTest/testHarness.sh

```diff
@@ -731,22 +764,54 @@ do
     continue
   fi
 
-  if [ "$verbose" == "off" ]
-  then
-    init=$testFile" ................................................................................................................."
-    init=${init:0:110}
-    printf "%03d/%d: %s " "$testNo" "$noOfTests" "$init"
-  else
-    printf "Running test %03d/%d: %s\n" "$testNo" "$noOfTests" "$testFile"
-  fi
-
   testNo=$testNo+1
   startDate=$(date)
   start=$(date --date="$startDate" +%s)
   endDate=""
+  typeset -i tryNo
+  tryNo=1
+
   if [ "$dryrun" == "off" ]
   then
-    runTest $testFile
+    while [ $tryNo -le $MAX_TRIES ]
+    do
+      if [ "$verbose" == "off" ]
+      then
+        tryNoInfo=""
+        if [ $tryNo != "1" ]
+        then
+           tryNoInfo="(intent $tryNo)"
+        fi
+
+        init=$testFile" ................................................................................................................."
+        init=${init:0:110}
+        printf "%03d/%d: %s %s " "$testNo" "$noOfTests" "$init" "$tryNoInfo"
+      else
+        printf "Running test %03d/%d: %s\n" "$testNo" "$noOfTests" "$testFile"
+      fi
+
+      runTest $testFile $tryNo
+      if [ $shellResult == "0" ]
+      then
+        if [ $tryNo != 1 ]
+        then
+          echo "OK"
```

Comment:
Yes - as you say.
The first 'informative line' is as before while the second and consecutive kines get an extra "(intent X).

Look up line 781:

    if [ $tryNo != 1 ]
      then
      tryNoInfo="(intent $tryNo)"